src/Framework/TestCase.php

```diff
@@ -1724,31 +1793,68 @@ private function handleDependencies(): bool
                     $dependency = $className . '::' . $dependency;
                 }
 
-                if (!isset($passedKeys[$dependency])) {
-                    $this->status = BaseTestRunner::STATUS_SKIPPED;
-
-                    $this->result->startTest($this);
-
-                    $this->result->addError(
-                        $this,
-                        new SkippedTestError(
-                            \sprintf(
-                                'This test depends on "%s" to pass.',
-                                $dependency
-                            )
-                        ),
-                        0
-                    );
+                if (\strpos($dependency, '\\') === 0) {
+                    $dependency = \substr($dependency, 1);
+                }
 
-                    $this->result->endTest($this, 0);
+                if (!isset($passedKeys[$dependency])) {
+                    if (
+                        !isset($skippedKeys[$dependency])
+                        && !isset($failedKeys[$dependency])
+                        && \strpos($dependency, '::') > 0
+                    ) {
+                        if ($className !== \explode('::', $dependency, 2)[0]) {
+                            $dependencyClass = \explode('::', $dependency, 2)[0];
+                            $dependencyKey   = self::DEPENDENCY_PREFIX . \ucfirst($dependencyClass);
+                            $oldErrorsCount  = self::$DEPENDENCY_TASK_RESULTS->errorCount();
+                            (new TestSuite($dependencyClass, $dependencyKey))->run(self::$DEPENDENCY_TASK_RESULTS);
```

Comment:
Before you put any more work into this, wait for a reply for @sebastianbergmann 

The design of this feature still worries me. Creating a new `TestSuite` for the missing dependenc(y|ies) and `run()` it _while running_ the main loop, is the least optimal way of solving this. If you want to run `@depends` before the dependant test, make sure the tests are in the optimal order before you start running.

- what protection is there against instantiating (and running) the same `TestSuite` twice?
- ...or more than twice? are results from the `@depends` test merged back into the 'main' `$testResult`
- how does all this work out when sent to the various loggers? Some of them, like JUNit, log the `TestSuite` hierarchy as the `start*/add*/end*` events are fired

Reply:
> what protection is there against instantiating (and running) the same TestSuite twice


It will be ran twice, if it is not in correct order. 

> ...or more than twice? are results from the @depends test merged back into the 'main'  $testResult

Not possible to run it over twice. The result will be put into dependency-only testResult in case it's required more than one time before it is ran in correct order.

> how does all this work out when sent to the various loggers? Some of them, like JUNit, log the TestSuite hierarchy as the start*/add*/end* events are fired

I'll test this. Should not affect that.